R/extract_features.R

```diff
@@ -453,24 +489,48 @@ input_fn_builder_EF <- function(features,
 #' @examples
 #' \dontrun{
 #' BERT_PRETRAINED_DIR <- download_BERT_checkpoint("bert_base_uncased")
-#' vocab_file <- file.path(BERT_PRETRAINED_DIR, 'vocab.txt')
-#' init_checkpoint <- file.path(BERT_PRETRAINED_DIR, 'bert_model.ckpt')
-#' bert_config_file <- file.path(BERT_PRETRAINED_DIR, 'bert_config.json')
-#' examples <- list(InputExample_EF(unique_id = 1,
-#'                                   text_a = "I saw the branch on the bank."),
-#'                  InputExample_EF(unique_id = 2,
-#'                                   text_a = "I saw the branch of the bank."))
-#' feats <- extract_features(examples = examples,
-#'                           vocab_file = vocab_file,
-#'                           bert_config_file = bert_config_file,
-#'                           init_checkpoint = init_checkpoint,
-#'                           batch_size = 2L)
+#' vocab_file <- file.path(BERT_PRETRAINED_DIR, "vocab.txt")
+#' init_checkpoint <- file.path(BERT_PRETRAINED_DIR, "bert_model.ckpt")
+#' bert_config_file <- file.path(BERT_PRETRAINED_DIR, "bert_config.json")
+#' examples <- list(
+#'   InputExample_EF(
+#'     unique_id = 1,
+#'     text_a = "I saw the branch on the bank."
+#'   ),
+#'   InputExample_EF(
+#'     unique_id = 2,
+#'     text_a = "I saw the branch of the bank."
+#'   )
+#' )
+#' feats <- extract_features(
+#'   examples = examples,
+#'   vocab_file = vocab_file,
+#'   bert_config_file = bert_config_file,
+#'   init_checkpoint = init_checkpoint,
+#'   batch_size = 2L
+#' )
 #' # can just specify checkpoint directory
-#' feats <- extract_features(examples = examples,
-#'                           ckpt_dir = BERT_PRETRAINED_DIR,
-#'                           batch_size = 2L)
+#' feats <- extract_features(
+#'   examples = examples,
+#'   ckpt_dir = BERT_PRETRAINED_DIR,
+#'   batch_size = 2L
+#' )
 #' }
 extract_features <- function(examples,
+                             model = c(
+                               "bert_base_uncased",
```

Comment:
just a thought, since this list is appearing more places: is there a way to refer to a master list in cases like this, so it's not hardcoded everywhere?

Reply:
Nothing I can find without getting reallllly fancy (or confusing; the easy thing would be to include them as data, but then users would have to load that data object to see them). I'll write a ticket to investigate it, but I *think* it'd involve adding functions that use them to a list somewhere, and then running a function that updates the formals of those functions. I can imagine building something like this into `factory`, though :)